internal/supported_types/aws_supported_types.go

```diff
@@ -0,0 +1,20 @@
+package supported_types
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+)
+
+func FilterAWSInstanceTypes(_ context.Context, it *[]clients.InstanceType) (*[]clients.InstanceType, error) {
+	for _, i := range *it {
+		_, ok := awsSupported[i.Name]
+		if ok {
+			// If the Instance type is present in the yml it is supported
+			i.Supported = true
+		}
+	}
+	// Returns all the Instance types received from aws
+	// if the instance type is supported the flag supported will be true otherwise false
+	return it, nil
+}
```

Comment:
This method should no longer return anything, you correctly use pointer to slice which means you can modify the slice as needed. Not even error is needed I think. Since it no longer filters anything, it could be named differently.

Please extract the `awsSupported[i.Name]` line and surrounding statements into a separate exported function something like `IsSupported`. We will need this in reservation endpoint check. This is ok to add in this PR as well, but we can do it separately if you prefer.